core/src/main/java/io/onetable/iceberg/IcebergClient.java

```diff
@@ -94,6 +96,26 @@ public IcebergClient(PerTableConfig perTableConfig, Configuration configuration)
       IcebergPartitionSpecSync partitionSpecSync,
       IcebergDataFileUpdatesSync dataFileUpdatesExtractor,
       IcebergTableManager tableManager) {
+    _init(
```

Comment:
This constructor should only be used for testing purposes and should have an appropriate annotation.

Reply:
That is outside the scope of the refactoring here, if wasn't there previously. If it wasn't there then it may be used externally and this would make this PR backward incompatible unnecessarily. Maybe we can make that a separate change?

Reply:
Done. We can always easily change it back if we need to.